pkg/eval/glob_test.go

```diff
@@ -88,3 +88,23 @@ func TestGlob_But(t *testing.T) {
 		That("put *[but:foobar][but:ipsum]").Puts("bar", "foo", "lorem"),
 	)
 }
+
+func TestGlob_Type(t *testing.T) {
+	_, cleanup := util.InTestDir()
+	defer cleanup()
+
+	mustMkdirAll("d1", "d2", ".d", "b/c")
+	mustCreateEmpty("bar", "foo", "ipsum", "lorem", "d1/f1", "d2/fm")
+
+	Test(t,
+		That("put **[d]").Puts("b/c", "b", "d1", "d2"),
+		That("put **[type:dir]").Puts("b/c", "b", "d1", "d2"),
+		That("put **[r]f*").Puts("d1/f1", "d2/fm", "foo"),
```

Comment:
This test doesn't really test the regular modifier since only regular files have `f` in their names, meaning that it fails to capture the bug I mentioned earlier in the review (where the rhs's nil TypeCb overrides the non-nil TypeCb).